binding/web/src/cobra.ts

```diff
@@ -228,15 +239,32 @@ export class Cobra {
       throw new CobraErrors.CobraInvalidArgumentError('Invalid AccessKey');
     }
 
+    let { device } = options;
+
+    const isSimd = await simd();
+    if (!isSimd) {
+      throw new CobraErrors.CobraRuntimeError('Browser not supported.');
+    }
+
+    const isWorkerScope = typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope;
+    if (!isWorkerScope) {
+      if (device && device !== "cpu:1") {
+        console.warn("Multi-threading is not supported on main thread.");
+      }
+      device = "cpu:1";
+    }
+
+    const sabDefined = (typeof SharedArrayBuffer !== 'undefined') && (device !== "cpu:1");
+
     return new Promise<Cobra>((resolve, reject) => {
       Cobra._cobraMutex
         .runExclusive(async () => {
-          const isSimd = await simd();
           const wasmOutput = await Cobra.initWasm(
             accessKey.trim(),
-            (isSimd) ? this._wasmSimd : this._wasm,
-            (isSimd) ? this._wasmSimdLib : this._wasmLib,
-            (isSimd) ? createModuleSimd : createModule);
+            (device) ? device : "best",
```

Comment:
After you make the updates above:
```suggestion
            device,
```